File: data/gcc_versions_comparison/src/program/src/func15.c
#include "program.h" 
array_t* func15(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array177;
   if (pCounter > 0) {
      array177 = vars->data[--pCounter];
      array177->refC++;
      DEBUG_COPY(array177->id);
   } else {
      array177 = (array_t*)malloc(sizeof(array_t));
      array177->size = 338;
      array177->refC = 1;
      array177->id = 177;
      array177->data = (unsigned int*)malloc(array177->size*sizeof(unsigned int));
      memset(array177->data, 0, array177->size*sizeof(unsigned int));
      DEBUG_NEW(array177->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array177;
   array_t* array178 = func18(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array178->id);
   free(params0.data);
   unsigned int loop51 = 0;
   unsigned int loopLimit51 = (rand()%loopsFactor)/2 + 1;
   for(; loop51 < loopLimit51; loop51++) {
      for (int i = 0; i < array177->size; i++) {
         array177->data[i]++;
      }
   }
   for (int i = 0; i < array177->size; i++) {
      if (array177->data[i] == 17) { 
         return array177;
      }
   }
   array178->refC--;
   if(array178->refC == 0) {
      free(array178->data);
      free(array178);
      DEBUG_FREE(array178->id);
   }
   return array177;
}

File: data/multilang_comparison/program_nim/src/func15.nim
proc func15*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array273: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array273 = vars.data[pCounter]
      inc array273.refC
      debugCopy(array273.id)
   else:
      array273 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array273.size = 685
      array273.refC = 1
      array273.id = 273
      array273.data = cast[ptr UncheckedArray[uint32]](alloc(array273.size * sizeof(uint32)))
      zeroMem(array273.data, array273.size * sizeof(uint32))
      debugNew(array273.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array273
   var array274: ptr ArrayT = func18(addr params0, getPath(), loopsFactor)
   debugReturn(array274.id)
   dealloc(params0.data)
   let loopLimit78 = (50) div (2) + 1
   for loop78 in 0 ..< loopLimit78:
      discard
      for i in 0 ..< array273.size:
         inc array273.data[i]
   for i in 0 ..< array274.size:
      if array274.data[i] == 5:
         return array274
   dec array274.refC
   if array274.refC == 0:
      dealloc(array274.data)
      dealloc(array274)
      debugFree(array274.id)
   return array273

